### PR TITLE
Downloads backwards compatible metadata

### LINKED ISSUE #523: Add backwards-compatibility test for being able to pull old data from a server
There's a backwards-compatibility test for loading old data and old key file formats.

There should also be one for serving the same fixture metadata and being able to pull with the new client.

### PR DESCRIPTION
**Pull request by cyli**
I had to re-check in the 0.1 stuff :|  I realized I needed the timestamp file in order to download.

I just reverted back to tag 0.1, `docker-compose build/up`, `make binaries`, and initialized a repo.  This time I included a changefile that added a target too, since I realized I forgot that last time.

Fixes #523

### CODE REVIEW

File: client/backwards_compatibility_test.go [hunk 1]
```diff
@@ -74,28 +76,62 @@ func Test0Dot1RepoFormat(t *testing.T) {
...
+
+	// targets should have 1 target, and it should be readable offline
+	targets, err := repo.ListTargets()
+	require.NoError(t, err)
+	require.Len(t, targets, 1)
+	require.Equal(t, "LICENSE", targets[0].Name)
+
+	// delete the timestamp metadata, since the server will ignore the uploaded
+	// one and try to create a new one from scratch, which will be the wrong version
+	assert.NoError(t, repo.fileStore.RemoveMeta(data.CanonicalTimestampRole))
```
Comment (endophage) [inline client/backwards_compatibility_test.go]: switch to `require`?

**Comment by endophage** [comment]
LGTM

File: client/backwards_compatibility_test.go [hunk 2]
```diff
@@ -14,7 +14,9 @@ import (
 	"github.com/docker/notary/client/changelist"
 	"github.com/docker/notary/passphrase"
 	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/store"
 	"github.com/stretchr/testify/assert"
```
Comment (riyazdf) [inline client/backwards_compatibility_test.go]: building off of @endophage's comment, are we ready to remove this import?

**Comment by riyazdf** [comment]
LGTM!

File: client/backwards_compatibility_test.go (hunk 1 above)
Comment (cyli) [inline client/backwards_compatibility_test.go]: Oops!  yes, will do that.

File: client/backwards_compatibility_test.go (hunk 2 above)
Comment (cyli) [inline client/backwards_compatibility_test.go]: Yep :)  Once I remove that assert.

File: client/backwards_compatibility_test.go [hunk 3]
```diff
@@ -74,28 +75,62 @@ func Test0Dot1RepoFormat(t *testing.T) {
...
+	require.NotEqual(t, oldTargetsKeys[0], newTargetsKeys[0])
+}
+
+// Ensures that the current client can download metadata that is published from notary 0.1 repos
+func TestDownloading0Dot1RepoFormat(t *testing.T) {
+	gun := "docker.com/notary0.1/samplerepo"
+	passwd := "randompass"
+
+	metaCache, err := store.NewFilesystemStore(
+		filepath.Join("../fixtures/compatibility/notary0.1/tuf", filepath.FromSlash(gun)),
```
Comment (diogomonica) [inline client/backwards_compatibility_test.go]: Won't this be expired eventually?

Comment (cyli) [inline client/backwards_compatibility_test.go]: Ah, right.  :|  I will check in a new version with expiry set to like 100 years from now.

Comment (diogomonica) [inline client/backwards_compatibility_test.go]: Sounds good.

**Comment by diogomonica** [comment]
LGTM after change in all the expirations of the fixture.

**Pull request merged**